### PR TITLE
tag policies not honored in reassignment after peer remove

### PR DESCRIPTION
**Pull request by tbeets**
Resolves issue in which tag policies (tag affinity and tag uniqueness constraint) were not honored when selecting a new peer for a stream RG to replace one just removed.  

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -4797,6 +4790,13 @@ func (cc *jetStreamCluster) selectPeerGroup(r int, cluster string, cfg *StreamCo
 			continue
 		}
 
+		// If ignore skip
+		if ip != nil {
```
Comment (derekcollison) [inline server/jetstream_cluster.go:4794]: nil maps ok to dereference. So you can just do line 4795 direct.

https://go.dev/play/p/XLBsPw7NUBM

**Review by derekcollison — APPROVED** [review]
LGTM - One small suggestion.

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -4769,6 +4753,13 @@ func (cc *jetStreamCluster) selectPeerGroup(r int, cluster string, cfg *StreamCo
 		}
 	}
 
+	// Map ignore
+	var ip map[string]struct{}
+	ip = make(map[string]struct{})
```
Comment (derekcollison) [inline server/jetstream_cluster.go:4758]: Should still conditionally create here.

File: server/jetstream_cluster.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster.go:4792]: This is where you can just dereference, and if nil will be ok.

**Review by derekcollison — CHANGES_REQUESTED** [review]

File: server/jetstream_cluster.go (hunk 2 above)
Comment (tbeets) [reply server/jetstream_cluster.go:4758]: Yep, just noticed.  Will fix.  Thx.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**